fern/products/sdks/guides/dynamic-authentication.mdx

```diff
@@ -0,0 +1,279 @@
+Your API may require dynamic authentication where credentials need to be generated or refreshed for each request, such as signing short-lived JWTs or rotating tokens. Fern-generated SDKs support this pattern through custom fetcher middleware.
+
+## Recommended: Custom fetcher middleware
+
+The best way to implement dynamic authentication in TypeScript SDKs is to use a custom fetcher. This acts as middleware for all requests, allowing you to inject authentication logic in a single place without overriding individual methods.
+
+### How it works
+
+When you enable `allowCustomFetcher` in your generator configuration, the generated SDK accepts a `fetcher` parameter in the client options. This fetcher wraps all HTTP requests, giving you a single injection point for authentication logic.
+
+### TypeScript example: Short-lived JWT signing
+
+Here's how to implement JWT signing with token memoization:
+
+<Steps>
+
+### Enable custom fetcher in generator configuration
+
+Add `allowCustomFetcher: true` to your `generators.yml`:
+
+```yaml title="generators.yml"
+default-group: local
+groups:
+  local:
+    generators:
+      - name: fernapi/fern-typescript-node-sdk
+        version: 0.x.x
+        output:
+          location: local-file-system
+          path: ../generated/typescript
+        config:
+          allowCustomFetcher: true
+```
+
+### Create a custom fetcher with JWT signing
+
+Create a fetcher function that wraps the default fetcher and injects JWT authentication:
+
+```typescript title="src/wrapper/jwtFetcher.ts"
+import * as jwt from "jsonwebtoken";
+import { fetcher as defaultFetcher, type FetchFunction } from "../core/fetcher";
+
+export function createJwtFetcher(privateKey: string): FetchFunction {
+  // Cache token to avoid regenerating on every request
+  let cachedToken: { value: string; expiresAt: number } | undefined;
+
+  return async (args) => {
+    const now = Math.floor(Date.now() / 1000);
+    
+    // Regenerate token if expired or about to expire (within 2 seconds)
+    if (!cachedToken || cachedToken.expiresAt - 2 <= now) {
+      const payload = {
+        iat: now,
+        exp: now + 15, // Token valid for 15 seconds
+      };
+      const token = jwt.sign(payload, privateKey, { algorithm: "RS256" });
+      cachedToken = { value: token, expiresAt: payload.exp };
+    }
+
+    // Inject JWT into request headers
+    const headers = {
+      ...(args.headers ?? {}),
+      Authorization: `Bearer ${cachedToken.value}`,
+    };
+
+    // Call the default fetcher with modified headers
+    return defaultFetcher({ ...args, headers });
+  };
+}
+```
+
+### Create a wrapper client
+
+Extend the generated client to use the custom fetcher:
+
+```typescript title="src/wrapper/PlantStoreClient.ts"
+import { PlantStoreClient as FernClient } from "../Client";
+import { createJwtFetcher } from "./jwtFetcher";
+
+export class PlantStoreClient extends FernClient {
+  constructor(options: { privateKey: string; environment: string }) {
```

Comment:
make sure you accept the same options as `FernClient`'s ctor does, minus the custom fetcher, and dynamically pass it along.